### PR TITLE
[v2] Remove mock imports

### PR DESCRIPTION
**Pull request by hssyoo**
Remove the `mock` dependency and replace all imports with `unittest.mock`.

This PR is a breakout PR of https://github.com/aws/aws-cli/pull/8757

### CODE REVIEW

File: tests/unit/customizations/codedeploy/test_push.py [hunk 1]
```diff
@@ -8,19 +8,18 @@
 #
 # or in the "license" file accompanying this file. This file is
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
-# ANY KIND, either express or implied. See the License for the specific
+# mock.ANY KIND, either express or implied. See the License for the specific
```
Comment (nateprewitt) [inline tests/unit/customizations/codedeploy/test_push.py:11]: Find and replace got a little too aggressive :D

```suggestion
# ANY KIND, either express or implied. See the License for the specific
```

**Review by nateprewitt — CHANGES_REQUESTED** [review]
Looks good, just one cleanup item.

File: tests/unit/customizations/codedeploy/test_push.py (hunk 1 above)
Comment (hssyoo) [reply tests/unit/customizations/codedeploy/test_push.py]: Lol nice catch

**Review by nateprewitt — APPROVED** [review]
:shipit:

**Pull request merged**